lib/compress/zstd_compress.c

```diff
@@ -2131,14 +2144,33 @@ static size_t ZSTD_resetCCtx_byCopyingCDict(ZSTD_CCtx* cctx,
                                                             : 0;
         size_t const hSize =  (size_t)1 << cdict_cParams->hashLog;
 
-        ZSTD_memcpy(cctx->blockState.matchState.hashTable,
-               cdict->matchState.hashTable,
-               hSize * sizeof(U32));
+        if (ZSTD_CDictIndicesAreTagged(&cctx->appliedParams.cParams)){
+            size_t i;
+            for (i = 0; i < hSize; i++) {
+                U32 const taggedIndex = cdict->matchState.hashTable[i];
+                U32 const index = taggedIndex >> ZSTD_SHORT_CACHE_TAG_BITS;
+                cctx->blockState.matchState.hashTable[i] = index;
+            }
+        } else {
+            ZSTD_memcpy(cctx->blockState.matchState.hashTable,
+                cdict->matchState.hashTable,
+                hSize * sizeof(U32));
+        }
```

Comment:
nit: Can you extract this out into a function that copies a table. E.g.

```
void ZSTD_copyCDictTableIntoCCtx(ZSTD_CCtx const* cctx, U32* dst, U32 const* src, size_t size);
```

And use it here and below.

Reply:
This will make it easier to measure future improvements from AVX2 and/or PREFETCHW. Good idea.